### ISSUE TITLE
[docs] Analyse docs taxonomy and propose new structure if required

### ISSUE THREAD

**Issue by rmoff** [treeverse/lakeFS#6197]
Current taxonomy has grown somewhat organically. 

![CleanShot_2023-07-12_at_16 11 02](https://github.com/treeverse/lakeFS/assets/3671582/caa22273-8f3f-4393-8446-ea551697739e)

It might be time for a bit of pruning and tidying of the branches. Analyse the current map, and propose a new one if needed.

**Comment by rmoff** [comment]
One of the canonical strategies for organising technical documentation is [Diátaxis](https://diataxis.fr). It's based on a map of *user needs* and breaks content down into four types across two different axes: 

![image](https://github.com/treeverse/lakeFS/assets/3671582/d752f7e9-7f56-4b67-b630-c514fdd88449)

We will use this as the guiding strategy for lakeFS documentation. Note that this is a *guide* and a *strategy*, it is not a dogmatic approach nor fixed structure to apply at all costs.

## tl;dr & Next Steps

We're going to update the lakeFS docs in an organic way towards an alignment with the [Diátaxis](https://diataxis.fr/) approach to documentation which classifies documentation into one of four types: 

* Tutorials
* How-To Guides
* Reference
* Explanation

**This will influence the navigation structure of the site as well as the way that page content is structured.** For notes and commentary please see below.

We will do this by iterating on the approach, starting with: 

* Consolidate some of the top-level elements into lower-elements
	* `Actions and Hooks` -> `How-To`
	* `Deploy and Setup` -> `How-To`
	* `Roadmap`, `Commitment to OSS`, `Contributing` -> Either `Understanding lakeFS` or new top level `The lakeFS Project`
	* `Use Cases`, `FAQ` -> Understanding lakeFS
* Dissipate `lakeFS Enterprise` and `lakeFS Cloud` docs out into the relevant sections (i.e. we won't have top-level elements for these)
* Move `Reference` down the left-nav order
* Iterate! 

---

## Implementation 

[According to the strategy](https://diataxis.fr/how-to-use-diataxis/#allow-your-work-to-develop-organically) we should make incremental improvements, not big-bang

> It’s natural to want to complete large tranches of work before you publish them, so that you have something substantial to show each time. Avoid this temptation - every step in the right direction is worth publishing immediately.

> Although Diátaxis is intended to provide a big picture of documentation, **don’t try to work on the big picture**. It’s both unnecessary and unhelpful. Diátaxis is designed to guide small steps; keep taking small steps to arrive where you want to go.

### [Complex hierarchies](https://diataxis.fr/complex-hierarchies/#two-dimensional-problems)

This is relevant for us as we decide how to present lakeFS, lakeFS Cloud, and lakeFS Enterprise documentation. 

> Diátaxis is not a scheme into which documentation must be placed - four boxes. It posit four different kinds of documentation, around which documentation should be structured, but this does not mean that there must be simply four divisions of documentation in the hierarchy, one for each of those categories.

> \[Diátaxis\] is not intended to be a plan,\[…\] It’s a guide, a map to help you check that you’re in the right place and going in the right directions.

> **Diátaxis is underpinned by attention to user needs**

> What we must document is the product _as it is for the user_, the product as it is in their hands and minds. (Sadly for the creators of products, how they conceive them is much less relevant.)

## Four Types of Content

Here are some key highlights on the four types, drawn from the [Diátaxis website](https://diataxis.fr).

### [Tutorials](https://diataxis.fr/tutorials/)

> A tutorial in other words is a lesson - a lesson concerned with learning **how** rather than learning **that**, because it’s concerned with skill: **practical, not theoretical knowledge**.
> 
> For a product, **a tutorial turns new learners into users**. An inadequate tutorial can prevent a project from acquiring new users.

> It’s also perfectly acceptable if what you get the beginner to do is not the way an experienced person would, or even if it’s not the ‘correct’ way - a tutorial for beginners is not the same thing as a manual for best practice.
> The point of a tutorial is to help your learner set out safely on their journey, not to get them to a final destination.

> **If the learner doesn’t need an explanation in order to complete the tutorial, don’t explain.**

> **Ignore options and alternatives**
> Your job is to guide the learner to a successful conclusion. There may be many interesting diversions along the way (different options for the command you’re using, different ways to use the API, different approaches to the task you’re describing) - ignore them. **Your guidance needs to remain focused on what’s required to reach the conclusion**, and everything else can be left for another time.
> Doing this helps keep your tutorial shorter and crisper, and saves both you and the reader from having to do extra cognitive work.

Examples of tutorials in the context of lakeFS could be: 

* Quickstart (existing)

**Be aware that Tutorials != How-To Guides**—see below

### [How-To Guides](https://diataxis.fr/how-to-guides/)

> A rich list of how-to guides is an encouraging suggestion of a product’s capabilities.

> If they’re well-written and address the right subjects, you’re likely to find that how-to guides are the most-read sections of your documentation.

> **How-to guides are wholly distinct from tutorials**. They are easily conflated, as both describe a series of practical steps that lead to the completion of some task. The user-needs that they serve are quite different however, and **conflating them is at the root of many difficulties that afflict documentation**.

> The problem or task is the concern of a how-to guide: **stick to that practical goal**. Anything else that’s added - unnecessary explanation, for example - distracts both you and the user and dilutes the useful power of the guide.

> **a how-to guide should not try to explain things.** Explanation here will simply get in the way of the action. If explanations are important, link to them.

Parallels between a [recipe and how-to](https://diataxis.fr/how-to-guides/#food-and-cooking)—you're telling the user the steps to take, not teaching them how to perform those steps.

Some of the existing lakeFS content that is a good example of How-To: 

* [Import data into lakeFS](https://docs.lakefs.io/howto/import.html)
* [Upgrading lakeFS](https://docs.lakefs.io/howto/upgrade.html)


### [Tutorials and How-To Guides](https://diataxis.fr/tutorials-how-to/#tutorials-how-to)

> Diátaxis insists that [what matters in documentation is the needs of the user](https://diataxis.fr/needs/#needs)…Sometimes the user is **at study**, and sometimes the user is **at work**. 
>
> **A tutorial serves the needs of the user who is at study**. Its obligation is _to provide a successful **learning experience**_. 
> **A how-to guide serves the needs of the user who is at work**. Its obligation is _to help the user **accomplish a task_**. 

In practice this is not always so crystal-clear, but [this table](https://diataxis.fr/tutorials-how-to/#understanding-the-distinction) provides a useful set of comparisons to help evaluate whether something is a guide or a how-to. 

tl;dr a tutorial is teaching something, whilst a how-to is how to get something done assuming an existing familiarity and understanding.

### [Reference](https://diataxis.fr/reference/)

> Reference material should be **austere and to the point**. One hardly _reads_ reference material; one _consults_ it.

> Although reference should not attempt to show how to perform tasks, it can and often needs to include a description of how something works or the correct way to use it.

> **Technical reference has one job: to describe**, and to do that clearly, accurately and comprehensively. Doing anything else - explaining, discussing, instructing, speculating - gets in the way of that job, and makes it harder for the reader to find the information they need.

> It can be tempting to introduce instruction and explanation, simply because technical reference can seem too bare. Instead, link to how-to guides, explanation and introductory tutorials as appropriate.

> **Examples** are valuable ways of providing illustration that helps readers understand reference, without becoming distracted from the job of describing. For example, an example of usage of a command can be a succinct way of illustrating it and its context.

### [Explanation](https://diataxis.fr/explanation/)

> Explanation is **discussion** that clarifies and illuminates a particular topic. Explanation is **understanding-oriented**.

> Explanation clarifies, deepens and broadens the reader’s understanding of a subject.

> Explanation, unlike the other three forms of documentation, doesn’t have a direct part in a user’s practice or work. This means that it’s sometimes seen as being of lesser importance. That’s a mistake; it may be less _urgent_ than the other three, but it’s no less _important_. It’s not a luxury. No practitioner of a craft can afford to be without an understanding of that craft, and needs the explanatory material that will help weave it together.

## Open questions

* How do you approach pages that have multiple types of content? e.g. [actions](https://docs.lakefs.io/hooks/) has explanation and reference. Do we carve that up into three small pages and reference each from the other? 
	* We perhaps reconcile the content types within the page and more clearly label them
	* Diátaxis doesn't necessarily mean physically carving up the pages
	* Looking closely at it there's no how-to—which suggests we should have a how-to on actions (could copy the example from lakeFS-samples)

## [lakeFS Guides](https://lakefs.io/guides/)
* [ref #285](https://github.com/treeverse/devex-internal/issues/285)
* These belong in docs
    > [What matters in documentation is the needs of the user](https://diataxis.fr/needs/#needs)
    * 👉🏻 Users will expect to find these guides under docs, not `lakeFS.io`
* We need to have a think about whether we're aiming for [tutorials](https://diataxis.fr/tutorials/) or [how-to guides](https://diataxis.fr/how-to-guides/) - a lift-and-shift of the existing content will not be suitable
* My [instinct](https://diataxis.fr/compass/#compass) is that these are how-to guides
* We would probably want to split some of the content out and link to it for things like the intros
	* For example, this [nice guide here about schema validation](https://lakefs.io/blog/schema-validation-lakefs-tutorial/) would have the `Introduction` section moved to `use cases` (or a new page under `Understanding lakeFS` or something like that) and the remaining docs page start from `Implementation of lakeFS for schema validation`

**Comment by iddoavn** [comment]
@rmoff - This sounds like a good idea to me (not that I'm an expert). Having said that - we should probably be careful with all the links we have today for documentation (Blogs, slack, website etc.) we might also have links from third party blogs etc. 
So maybe there could be a way to change this with redirects etc. to avoid breaking old links?

**Comment by rmoff** [comment]
@iddoavn yes I've included redirects for all of the pages that move :)
